x-pack/plugins/alerting/server/rules_client/methods/bulk_edit.ts

```diff
@@ -540,23 +546,28 @@ async function getUpdatedAttributesFromOperations(
     // the `isAttributesUpdateSkipped` flag to false.
     switch (operation.field) {
       case 'actions': {
+        const updatedOperation = {
+          ...operation,
+          value: addUuid(operation.value),
+        };
```

Comment:
What is the intended scope of uniqueness for the `uuid` field? Is it a rule, all rules in a space, all rules in a cluster, all rules across all clusters?

Here, when bulk adding the same action to multiple rules, the `RulesClient` assigns the same generated value to `uuid` in every rule, per action, of course. Which means the scope of uniqueness looks to be a given rule. Is this correct understanding and is that how you folks think it should be working?

Reply:
Scope is the rule.